whatshap/cli/haplotagphase.py

```diff
@@ -0,0 +1,351 @@
+                genotypes = variant_table.genotypes_of(sample)
+                homozygous = dict()
+                change = dict()
+                phased = dict()
+                homozygous_number = 0
+                phased_number = 0
+                for variant, (phase, genotype) in zip(
+                    variant_table.variants, zip(phases, genotypes)
+                ):
+                    homozygous[variant.position] = genotype.is_homozygous()
+                    phased[variant.position] = phase
+                    phased_number += phase is not None
+                    homozygous_number += genotype.is_homozygous()
+                    change[variant.position] = variant
+                logger.info(f"Number of homozygous variants is {homozygous_number}")
+                logger.info(f"Number of already phased variants is {phased_number}")
+                with timers("compute-votes"):
+                    votes = compute_votes(homozygous, reads)
+                with timers("compute-consensus"):
+                    sample_to_super_reads[sample], sample_to_components[sample] = consensus(
+                        only_indels, gap_threshold, cut_poly, fasta_chr, change, phased, votes
+                    )
+            with timers("write-vcf"):
+                vcf_writer.write(chromosome, sample_to_super_reads, sample_to_components)
+    timers.stop("haplotagphase-run")
+    log_time_and_memory_usage(timers)
+
+
+def log_time_and_memory_usage(timers):
+    logger.info("\n# Resource usage")
+    log_memory_usage()
+    # fmt: off
+    logger.info("Finished in :                              %6.1f s", timers.elapsed("haplotagphase-run"))
+    logger.info("Time spent reading reference:              %6.1f s", timers.elapsed("read-fasta"))
+    logger.info("Time spent reading VCF:                    %6.1f s", timers.elapsed("parse-vcf"))
+    logger.info("Time spent writing VCF:                    %6.1f s", timers.elapsed("write-vcf"))
+    logger.info("Time spent reading BAM:                    %6.1f s", timers.elapsed("read-bam"))
+    logger.info("Time spent computing votes:                %6.1f s", timers.elapsed("compute-votes"))
+    logger.info("Time spent spent computing consensus:      %6.1f s", timers.elapsed("compute-consensus"))
+    # fmt: on
+
+
+def consensus(
+    only_indels: bool,
+    gap_threshold: int,
+    cut_poly: int,
+    fasta_chr: str,
+    change: Dict[int, VcfVariant],
+    phased: Dict[int, Optional[VariantCallPhase]],
+    votes: Dict[int, Dict[Tuple[int, int], int]],
+) -> Tuple[List[List[Read]], Dict[int, int]]:
+    """
+    Compute a consensus based on voting and filtering criteria.
+
+    This function processes variant votes to create two consensus sequences (super reads),
+    taking into account phasing information, indel preference, gap threshold, and homopolymer
+    length.
+
+    Args:
+        - only_indels (bool): If True, only consider indels for inclusion in the consensus. SNVs are ignored.
```

Comment:
There should be no leading dashes, see [example](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#overview).